src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.TimeDate/Components/ResultHelper.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Windows;
+using Microsoft.PowerToys.Run.Plugin.TimeDate.Properties;
+using Wox.Plugin.Logger;
+
+namespace Microsoft.PowerToys.Run.Plugin.TimeDate.Components
+{
+    public static class ResultHelper
+    {
+        /// <summary>
+        /// Returns a list with all available commands
+        /// </summary>
+        /// <param name="isKeywordSearch">Is this a search with plugin activation keyword or not</param>
+        /// <param name="timeLong">Required for UnitTest: Show time in long format</param>
+        /// <param name="dateLong">Required for UnitTest: Show date in long format</param>
+        /// <param name="timestamp">Required for UnitTest: Use custom <see cref="DateTime"/> object to calculate results</param>
+        /// <returns>List of results</returns>
+        public static List<AvailableResult> GetCommandList(bool isKeywordSearch, bool? timeLong = null, bool? dateLong = null, DateTime? timestamp = null)
+        {
+            List<AvailableResult> results = new List<AvailableResult>();
+            DateTime dateTimeNow = timestamp ?? DateTime.Now;
+            DateTime dateTimeNowUtc = dateTimeNow.ToUniversalTime();
+            bool timeExtended = timeLong ?? TimeDateSettings.Instance.TimeWithSeconds;
+            bool dateExtended = dateLong ?? TimeDateSettings.Instance.DateWithWeekday;
+
+            results.AddRange(new[]
+            {
+                new AvailableResult()
+                {
+                    Value = dateTimeNow.ToString(TimeAndDateHelper.GetStringFormat(FormatStringType.Time, timeExtended, dateExtended)),
+                    Label = Resources.Microsoft_plugin_timedate_time,
+                    IconType = ResultIconType.Time,
+                },
+                new AvailableResult()
+                {
+                    Value = dateTimeNow.ToString(TimeAndDateHelper.GetStringFormat(FormatStringType.Date, timeExtended, dateExtended)),
+                    Label = Resources.Microsoft_plugin_timedate_date,
+                    IconType = ResultIconType.Date,
+                },
+                new AvailableResult()
+                {
+                    Value = dateTimeNow.ToString(TimeAndDateHelper.GetStringFormat(FormatStringType.DateTime, timeExtended, dateExtended)),
+                    Label = timestamp == null ? Resources.Microsoft_plugin_timedate_Now : Resources.Microsoft_plugin_timedate_DateAndTime,
+                    IconType = ResultIconType.DateTime,
+                },
+            });
+
+            if (isKeywordSearch || !TimeDateSettings.Instance.OnlyDateTimeNowGlobal)
+            {
+                long unixTimestamp = (long)dateTimeNowUtc.Subtract(new DateTime(1970, 1, 1)).TotalSeconds;
```

Comment:
// Used long instead of int because int is to small after `03:14:07 UTC 2038-01-19`